src/vector/geometry.rs

```diff
@@ -310,6 +312,13 @@ impl Geometry {
         Geometry::with_c_geometry(c_geom, false)
     }
 
+    /// Get a reference to the geometry at given `index`
+    pub fn get_geometry(&self, index: usize) -> Result<GeometryRef> {
+        let geom = unsafe { self.get_unowned_geometry(index) };
+        let gref = GeometryRef { geom, _lifetime: PhantomData::default() };
+        Ok(gref)
```

Comment:
Just wondering why you're returning a Result here when there doesn't seem to be any circumstance under which it could be an Err